codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/Instantiator.kt

```diff
@@ -49,57 +51,48 @@ import software.amazon.smithy.rust.codegen.core.smithy.rustType
 import software.amazon.smithy.rust.codegen.core.util.dq
 import software.amazon.smithy.rust.codegen.core.util.expectMember
 import software.amazon.smithy.rust.codegen.core.util.hasTrait
-import software.amazon.smithy.rust.codegen.core.util.isStreaming
 import software.amazon.smithy.rust.codegen.core.util.letIf
 
 /**
- * Instantiator generates code to instantiate a given Shape given a `Node` representing the value
+ * Instantiator generates code to instantiate a given Shape given a `Node` representing the value.
  *
- * This is primarily used during Protocol test generation
+ * This is only used during protocol test generation.
  */
-class Instantiator(
+open class Instantiator(
     private val symbolProvider: RustSymbolProvider,
     private val model: Model,
     private val runtimeConfig: RuntimeConfig,
-    private val target: CodegenTarget,
+    /**
+     * A function that given a symbol for an enum shape and a string, returns a writable to instantiate the enum with
+     * the string value.
+     **/
+    private val enumFromStringFn: (Symbol, String) -> Writable,
```

Comment:
It's not immediately clear to me why this would be configurable (i.e, different in the client vs the server).
Maybe you could add a comment about that either here or on `ClientInstatiator::enumFromStringFn`/`ServerInstatntiator::enumFromStringFn`?

Reply:
https://github.com/awslabs/smithy-rs/pull/1863/commits/7b7bf6ce024bf313b11bfc20c9101614c667eb04